craft_parts/infos.py

```diff
@@ -47,16 +52,25 @@ def __init__(
         self,
         *,
         application_name: str = utils.package_name(),
+        cache_dir: Union[Path, str] = "",
         arch: str = "",
         base: str = "",
         parallel_build_count: int = 1,
         project_dirs: ProjectDirs = None,
         **custom_args,  # custom passthrough args
     ):
+        if cache_dir:
+            cache_path = Path(cache_dir)
+        else:
+            cache_path = BaseDirectory.save_cache_path(
```

Comment:
why not just fail or require it?

Reply:
I think it'd best to require it too. Maybe set a default parameter?

At the very least, use `None` instead of `""` for optional, if being kept.  :D